package/MDAnalysis/coordinates/chemfiles.py

```diff
@@ -308,8 +309,7 @@ def __init__(
             raise RuntimeError(
                 "Please install Chemfiles > {}" "".format(MIN_CHEMFILES_VERSION)
             )
-        self.filename = filename
-        self.n_atoms = n_atoms
+        super().__init__(filename, n_atoms, append)
         if mode != "a" and mode != "w":
             raise IOError("Expected 'a' or 'w' as mode in ChemfilesWriter")
```

Comment:
ChemFiles is possibly using `'a'` for append mode here

Reply:
It's up for discussion but my idea is to focus this PR on only adding general API to WriterBase and leave specific implementations to other PRs (in a batch for similar Writers) so it doesn't get messy :)

Also after this PR is merged, the current API for `mode=a` (if anyone is really using it) should not break.

Reply:
I support @yuxuanzhuang's approach here, keep it simple to make sure we can merge quickly.